src/windows.rs

```diff
@@ -0,0 +1,463 @@
+                {
+                    device_number += 1;
+                    display_devices.push(device.clone());
+                }
+                if display_devices.len() != physical_monitors.len() {
+                    // There doesn't seem to be anyway to directly associate a physical monitor handle
+                    // with the equivalent display device, other than by array indexing
+                    // https://stackoverflow.com/questions/63095216/how-to-associate-physical-monitor-with-monitor-deviceid
+                    return vec![Err(SysError::EnumerationMismatch)];
+                }
+
+                physical_monitors
+                    .into_iter()
+                    .zip(display_devices)
+                    .map(|(physical_monitor, mut display_device)| {
+                        // Get a file handle for this physical monitor
+                        // Note this is a different type of handle
+                        let handle = CreateFileW(
+                            PWSTR(display_device.DeviceID.as_mut_ptr()),
+                            FILE_ACCESS_FLAGS(GENERIC_READ | GENERIC_WRITE),
+                            FILE_SHARE_MODE(0),
+                            std::ptr::null_mut(),
+                            OPEN_EXISTING,
+                            FILE_FLAGS_AND_ATTRIBUTES(0),
+                            HANDLE::NULL,
+                        );
+                        if handle.is_invalid() {
+                            let e = GetLastError();
+                            // This error occurs for virtual devices e.g. Remote Desktop
+                            // sessions - they are not real monitors
+                            if e == ERROR_ACCESS_DENIED {
+                                return None;
+                            }
+                            return Some(Err(SysError::ListingDevicesFailed("CreateFileW", e)));
+                        }
+
+                        Some(Ok(Brightness {
+                            physical_monitor: physical_monitor.hPhysicalMonitor,
+                            file_handle: handle,
+                            device_name: wchar_to_string(&display_device.DeviceName),
+                            device_description: wchar_to_string(&{
+                                physical_monitor.szPhysicalMonitorDescription
+                            }),
+                            device_string: wchar_to_string(&display_device.DeviceString),
+                            device_id: wchar_to_string(&display_device.DeviceID),
+                            device_key: wchar_to_string(&display_device.DeviceKey),
+                        }))
+                    })
+                    .map(|e| e.into_iter())
+                    .flatten()
+                    .collect()
+            }))
+        }
+    };
+    futures::stream::iter(devices)
+}
+
+#[derive(Debug, Error, Clone)]
+pub enum SysError {
+    #[error("Call to {0}() failed, GetLastError was {1:?}")]
```

Comment:
Most variants have nearly identical messages that indicate which Win API function failed without more context as to what higher-level operation was attempted, even though the variant name has this information.

I think it would be more informative to define one variant per failing operation/call such that:
- The variant error message is a short description of the error or what failed, e.g. "Enumerating monitors failed".
- The variant contains a `source: windows::Error` field that becomes part of the chained list of errors and whose `Display` implementation should provide a more human-friendly description of the underlying error instead of an error code.

[Example](https://github.com/stephaneyfx/brightness/blob/16489a66ac1bb9f5cfa33786566600f95076f8e5/src/windows.rs#L214)

Reply:
done